### DISCUSSION TITLE
Feature request: expose newEvent function to Lua

### DISCUSSION THREAD

**Discussion by ghost** [godlikepanos/anki-3d-engine#105] [Ideas]
Afaict, `EventManager` exposes only `newLightEvent`, but it would be cool if it would expose `newEvent` as well, so one could pass a string with Lua code for `update` and `onKilled` functions from the main Lua script, what do you think? :)

Thanks!

**Comment by godlikepanos** [comment]
Hi. In C++ the `EventManager::newEvent` is a templated method and I'm not aware of a way to expose it in Lua directly. What I did for the SceneGraph which is similar (`SceneGraph::newSceneNode<>`) was to expose in Lua a number of specialized methods. For example in Lua there is `SceneGraph:newModelNode` which translates to C++ as `SceneGraph::newSceneNode<ModelNode>`.

I exposed a few more things related to events in commit 5d2f7def51c7bfcd520deec5e60cacb12f669418. Please let me know if that works

**Reply by ghost** [reply]
Awesome! `newScriptEvent` is exactly what I wanted. Thanks! :)
